pkg/controller/worker/machines.go

```diff
@@ -143,11 +153,16 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 		generateMachineClassAndDeployment := func(zone *zoneInfo, machineSet *machineSetInfo, subnetName, workerPoolHash string) (worker.MachineDeployment, map[string]interface{}) {
 			var (
 				machineDeployment = worker.MachineDeployment{
-					Minimum:              pool.Minimum,
-					Maximum:              pool.Maximum,
-					MaxSurge:             pool.MaxSurge,
-					MaxUnavailable:       pool.MaxUnavailable,
-					Labels:               pool.Labels,
+					Minimum:        pool.Minimum,
+					Maximum:        pool.Maximum,
+					MaxSurge:       pool.MaxSurge,
+					MaxUnavailable: pool.MaxUnavailable,
+					Labels: func() map[string]string {
+						if csiEnabled && zone != nil {
+							return utils.MergeStringMaps(pool.Labels, map[string]string{azureCSIDiskDriverTopologyKey: zone.name})
+						}
+						return pool.Labels
+					}(),
```

Comment:
Can we move this out to a dedicated function?
I'm not a big fan of using closures/anonymous functions if avoidable.
Bit just a nit other than this it looks good to me.

Reply:
Sure, will do that.